src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,78 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private String lineSeparator = System.lineSeparator();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        if (validateInput(names, data, dateFrom, dateTo)) {
+            throw new DataFormatException("Data is empty");
+        }
+
+        LocalDate[] dataDate = new LocalDate[data.length];
+        String[] dataName = new String[data.length];
+        String[] dataHours = new String[data.length];
+        String[] dataSalaryPerHour = new String[data.length];
+
+        try {
+            for (int i = 0; i < data.length; i++) {
+                String[] dataSplit = data[i].split(" ");
+                dataDate[i] = LocalDate.parse(dataSplit[0], FORMATTER);
+                dataName[i] = dataSplit[1];
+                dataHours[i] = dataSplit[2];
+                dataSalaryPerHour[i] = dataSplit[3];
+            }
+        } catch (DateTimeParseException e) {
+            throw new DataFormatException("Wrong date format");
+        }
+
+        LocalDate dateFromLocalDate = parseDateFrom(dateFrom);
+        LocalDate dateToLocalDate = parseDateTo(dateTo);
+
+        StringBuilder result = new StringBuilder();
+        result.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+
+        for (String name : names) {
+            int sum = 0;
+            for (int i = 0; i < data.length; i++) {
+                if (name.equals(dataName[i])
+                        && dataDate[i].compareTo(dateFromLocalDate) >= 0
+                        && dataDate[i].compareTo(dateToLocalDate) <= 0) {
+                    sum += Integer.valueOf(dataHours[i]) * Integer.valueOf(dataSalaryPerHour[i]);
+                }
+            }
```

Comment:
here we should add logic to parse data[] record to name/hours/salaryPerHour after removal of additional arrays